includes/assets/js/NFDPluginsMarketplace.js

```diff
@@ -1,99 +1,128 @@
+    );
+    filterArea.parentNode.insertBefore(
+      pluginsListTable,
+      filterArea.nextSibling
+    );
+  }
+
+  // Filter products to only show plugins
+  filterProducts(products) {
+    const plugins = products.filter((product) => product.type == 'plugin');
+
+    return plugins;
+  }
+
+  // Returns whether the environment supports CTB
+  supportsCTB(products) {
+    if (ctbAvailableProducts.includes(products?.clickToBuyId)) return true;
+    if (typeof nfdctb === 'undefined') {
+      return false;
     }
 
-     // Returns whether the environment supports CTB
-    supportsCTB() {
-        if (typeof nfdctb === 'undefined') {
-            return false;
-        }
-
-        return nfdctb.supportsCTB;
-    }
-
-    // Show the filtered products on the page
-    showProducts(products) {
-        
-        // Helper function to render the thumbnail for each product
-        const renderThumbnail = (product) => {
-            if (product.smallThumbnailUrl) {
-                return `<img src="${product.smallThumbnailUrl}" alt="${product.name} Thumbnail" width="128px" height="128px">`;
-            }
-
-            const fallback = 'https://hiive-space-cdn.nyc3.cdn.digitaloceanspaces.com/marketplace/premium-plugins/fallback.svg';
-            return `<img src="${fallback}" alt="${product.name} Thumbnail" width="128px" height="128px">`;
-        }
-
-        // Helper function to render the action buttons for each product
-        const renderActions = (product) => {
-            const supportsCTB = this.supportsCTB();
-            let actionButtons = [];
-
-            if (product.primaryUrl && product.price) {
-                const primaryAction = `<a target="_blank" 
+    return nfdctb.supportsCTB;
+  }
+
+  // Show the filtered products on the page
+  showProducts(products) {
+    // Helper function to render the thumbnail for each product
+    const renderThumbnail = (product) => {
+      if (product.smallThumbnailUrl) {
+        return `<img src="${product.smallThumbnailUrl}" alt="${product.name} Thumbnail" width="128px" height="128px">`;
+      }
+
+      const fallback =
+        'https://hiive-space-cdn.nyc3.cdn.digitaloceanspaces.com/marketplace/premium-plugins/fallback.svg';
+      return `<img src="${fallback}" alt="${product.name} Thumbnail" width="128px" height="128px">`;
+    };
+
+    // Helper function to render the action buttons for each product
+    const renderActions = (product) => {
+      const supportsCTB = this.supportsCTB(product);
+      let actionButtons = [];
+
+      if (product.primaryUrl && product.price) {
+        const primaryAction = `<a target="_blank" 
                 href="${product.primaryUrl}" 
                 class="button button-primary nfd-ctb-btn"
-                ${supportsCTB && product.clickToBuyId ? 'data-action="load-nfd-ctb" data-ctb-id="' + product.clickToBuyId + '"' : ''}>Buy $${product.price}</a>`;
+                ${
+                  supportsCTB && product?.clickToBuyId
```

Comment:
Here we can safely add the ctb attributes and don't need to check the supportsCTB value.

Reply:
done